packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -563,6 +570,21 @@ class HotRunner extends ResidentRunner {
     );
   }
 
+  bool _isPaused() {
+    for (FlutterDevice device in flutterDevices) {
+      for (FlutterView view in device.views) {
+        if (view.uiIsolate != null) {
+          final ServiceEvent pauseEvent = view.uiIsolate.pauseEvent;
+          if (pauseEvent != null && pauseEvent.isPauseEvent) {
```

Comment:
I think the name is a bit misleading - from the service protocol docs:
```
// The last pause event delivered to the isolate. If the isolate is
// running, this will be a resume event.
Event pauseEvent;
```